Add basic saved team support

magical opened the pull request:
Adds persistence for teams, backed by an SQLite database.  Only supports a single team for now, though there is some skeletal support for multiple teams.

Bug: the add/remove button PokemonItemDetailView needs to be updated to reflect whether the pokemon is a current team member.

magical committed:
Add basic classes for team persistence

magical committed:
Move team database stuff to new 'db' subpackage

magical committed:
[wip] Wire database storage up to team and pokemon detail views

Untested; probably doesn't work yet.

magical committed:
Merge branch 'master' into savedteam

magical committed:
Merge branch 'master' into savedteam

magical committed:
Get adding and removing team members basically sort of working

magical committed:
It helps if you call the right method

magical committed:
fix some more null pointer defererences, maybe?

magical committed:
hack hack hack viewModel.getPokemonById

This definitely isn't right, but it doesn't crash, usually.

magical committed:
Fix getPokemonById

magical committed:
Show add/remove button in pokemon activity when navigating from the team view

magical committed:
Remove unused variable

magical committed:
Remove redundant build lines

mwyoung reviewed (approved):
I currently don't have any issues with this code.

hammockt commented:
Are you still experiencing the loading bug?

Also just curious but do you prefer accessing live data with transformations? Since I am implementing the repository I would rather create a interface that you guys would prefer to work with. I thought methods in the PokeAPIViewModel would be the preferred way to access the internal data, but it is whatever you guys are more comfortable with.

magical committed:
Rename DBUtils to TeamUtils

magical committed:
Add doc comment for TeamUtils

magical commented:
@hammockt Yep, as we discussed in person, https://github.com/OSU-CS492-W19/final-project-team-pikachu/pull/18/commits/cb965a14f6981fdf60af6563f7f52987b639e067 fixed all the issues i was having with loading.

I also renamed DBUtils to TeamUtils and added a TODO about turning it into a ViewModel later.
